ingestion/src/metadata/ingestion/source/storage/s3/metadata.py

```diff
@@ -98,10 +101,16 @@ def get_containers(self) -> Iterable[S3ContainerDetails]:
                     bucket_response=bucket_response
                 )
                 self._bucket_cache[bucket_response.name] = self.context.container
-
-                metadata_config = self._load_metadata_file(
-                    bucket_name=bucket_response.name
-                )
+                try:
+                    metadata_config_response = self.s3_reader.read(
```

Comment:
can we keep this logic inside the `_load_metadata_file`? OK to restructure it but we can just "hide" this logic there